virttest/utils_net.py

```diff
@@ -1185,23 +1185,21 @@ def get_stp_status(self, brname):
 __bridge = Bridge()
 
 
-def __init_openvswitch(func):
+def __init_openvswitch():
     """
-    Decorator used for late init of __ovs variable.
+    Initialize the __ovs variable.
     """
-    def wrap_init(*args, **kargs):
-        global __ovs
-        if __ovs is None:
-            try:
-                __ovs = factory(openvswitch.OpenVSwitchSystem)()
-                __ovs.init_system()
-                if (not __ovs.check()):
-                    raise Exception("Check of OpenVSwitch failed.")
-            except Exception as e:
-                LOG.debug("Host does not support OpenVSwitch: %s", e)
-
-        return func(*args, **kargs)
-    return wrap_init
+    global __ovs
+    if __ovs is None:
+        try:
+            __ovs = factory(openvswitch.OpenVSwitchSystem)()
+            __ovs.init_system()
+            if (not __ovs.check()):
+                raise Exception("Check of OpenVSwitch failed.")
+        except Exception as e:
+            LOG.error("Host does not support OpenVSwitch: %s", e)
+            __ovs = None
+    return __ovs
```

Comment:
As `__ovs` be defined as a global parameter,   how about delete `return __ovs`, and call `__init_openvswitch` directly without return value ?

Reply:
Considering the situation that there would likely be further improvements to the network related modules, it is better to not use the global var directly.

Reply:
Don't use global var is make sense to me.   So with the new change, remove decorator, the definition can be change too i think.

Reply:
Ack with this, thanks!